src/win/thread.c

```diff
@@ -182,6 +182,7 @@ int uv_thread_join(uv_thread_t *tid) {
   else {
     CloseHandle(*tid);
     *tid = 0;
+    MemoryBarrier();  /* For feature parity with pthread_join(). */
```

Comment:
It's vaguely documented here https://msdn.microsoft.com/en-us/library/ms686355.aspx:
"The following synchronization functions use the appropriate barriers to ensure memory ordering: ... Wait functions"